src/sql/workbench/contrib/notebook/browser/cellViews/cellToolbar.component.ts

```diff
@@ -14,13 +22,78 @@ export const CELL_TOOLBAR_SELECTOR: string = 'cell-toolbar-component';
 	templateUrl: decodeURI(require.toUrl('./cellToolbar.component.html'))
 })
 export class CellToolbarComponent {
+	@ViewChild('celltoolbar', { read: ElementRef }) private celltoolbar: ElementRef;
+
 	public buttonEdit = localize('buttonEdit', "Edit");
 	public buttonClose = localize('buttonClose', "Close");
 	public buttonAdd = localize('buttonAdd', "Add new cell");
-	public buttonMoveDown = localize('buttonMoveDown', "Move cell down");
-	public buttonMoveUp = localize('buttonMoveUp', "Move cell up");
 	public buttonDelete = localize('buttonDelete', "Delete cell");
+	public buttonMoreActions = localize('buttonMoreActions', "More actions");
+
+	@Input() public cellModel: ICellModel;
+	@Input() set model(value: NotebookModel) {
+		this._model = value;
+	}
+	private _actionBar: Taskbar;
+	private _model: NotebookModel;
+	private _editCellAction: EditCellAction;
+
+	constructor(
+		@Inject(IInstantiationService) private _instantiationService: IInstantiationService,
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService
+	) {
+	}
+
+	get model(): NotebookModel {
+		return this._model;
+	}
+
+	ngOnInit() {
+		this.initActionBar();
+	}
+
+	private initActionBar(): void {
+		let addCodeCellButton = new AddCellAction('notebook.AddCodeCell', localize('codePreview', "Code cell"), 'notebook-button masked-pseudo code');
+		addCodeCellButton.cellType = CellTypes.Code;
+
+		let addTextCellButton = new AddCellAction('notebook.AddTextCell', localize('textPreview', "Markdown cell"), 'notebook-button masked-pseudo markdown');
+		addTextCellButton.cellType = CellTypes.Markdown;
+
+		let deleteButton = this._instantiationService.createInstance(CellToolbarAction, 'notebook.deleteCell', '', 'codicon masked-icon delete', this.buttonDelete, this.cellModel);
+
+		// Todo: Get this to show the list of actions specific to code or markdown cell.
+		let moreActionsButton = new MoreActions('notebook.moreActions', this.buttonMoreActions, 'codicon masked-icon more');
+
+		// Todo: Wire up toolbarToggleEditMode
+		// Todo: Wire up toolbarUnselectActiveCell
+		this._editCellAction = this._instantiationService.createInstance(EditCellAction, 'notebook.editCell', true);
+		this._editCellAction.enabled = true;
+
+		let taskbar = <HTMLElement>this.celltoolbar.nativeElement;
+		this._actionBar = new Taskbar(taskbar);
+		this._actionBar.context = this;
+
+		let buttonDropdownContainer = DOM.$('li.action-item');
+		buttonDropdownContainer.setAttribute('role', 'presentation');
+		let dropdownMenuActionViewItem = new DropdownMenuActionViewItem(
+			addCodeCellButton,
+			[addCodeCellButton, addTextCellButton],
```

Comment:
Here I have includes just two actions. If you look at the current _more actions_ ellipses in production, you see the four insert options (below image). I don't know how to handle the insert above and insert below. 
<img width="188" alt="image" src="https://user-images.githubusercontent.com/12279161/83060402-41ad0380-a010-11ea-8b92-25549c826263.png">

Reply:
Not sure what you mean here by "how to handle". Do you mean how to display them in the new toolbar or something else?

Reply:
Take a look at `AddCellFromContextAction`

Reply:
I borrowed most of the code from cellToggleMoreActions.ts and revised it to better suit the context of the cell toolbar.